### ISSUE TITLE
Suggested revisions in v2's L2

### ISSUE THREAD

**Issue by harigopal** [pupilfirst/wd201-scripts#210]
## TL;DR

1. Instead of demonstrating a command-line app that accepts data via `STDIN`, let's demo a similar app that accepts data via command-line arguments. Start simple with a single unnamed argument.
2. Increase complexity of command-line arguments (add more args), and demonstrate how third-party _packages_ let us deal with complex requirements without writing our own code by introducing something like [minimist](https://www.npmjs.com/package/minimist).
3. For the milestone task, let's ask the user to supply the _port_ which the server listens on using the `--port=` or `-p=` flags.
4. The milestone task's text should also be much clearer about what it expects students to do. The use of `registration.html` from WD101 reads as a suggestion right now - this should be written as a necessary step because that's a part of the test specification.

## Reasons for these changes

Multiple students have expressed that the L2 milestone is confusing. I looked at both the lessons and the milestone, and agree. The milestone text seems to imply that the use of `registration.html` is optional, but the test specifications indicate that as a requirement. Also, given that the `registration.html` is _right there_ in the repo alongside the script, it doesn't make much sense to input the path to that file via `STDIN`. This also complicates the testing of submissions because interacting with apps via `STDIN` is usually a pain.

While going through the level, I also noticed that the level that deals with `npm` doesn't actually have students installing any package. This seems like an oversight since the main point of NPM is the package ecosystem. We really need to talk about why the package ecosystem is important, and talk about how we'll often search for solutions to common problems by searching for a package. This is where the idea of starting simple, and then introducing minimist came from.

This also solves the problem of testing via `STDIN`, and the vagueness around how to verify whether test input was accepted or not. If `--port` is accepted via the command-line, the test is simply whether there's a server listening on the port - it should be way easier to implement using Jest. Also, accepting data using CLI arguments is probably more common than using `STDIN` - it's likely a better data input method to introduce students to.

**Comment by reenas** [comment]
@vikispeaks Is there any update on this?

**Comment by vikispeaks** [comment]
Working on this incorporating some suggestions @mjzac provided. Will share MR today.

**Comment by vikispeaks** [comment]
@harigopal @reenas Fixed with https://github.com/pupilfirst/wd201-scripts/pull/220. Please review.
